corehq/apps/locations/models.py

```diff
@@ -374,6 +397,15 @@ def by_location_id(cls, location_id):
         except cls.DoesNotExist:
             return None
 
+    def linked_supply_point(self):
+        from corehq.apps.commtrack.models import SupplyPointCase
+        if not self.supply_point_id:
+            return None
+        try:
+            return SupplyPointCase.get(self.supply_point_id)
+        except:
+            return None
```

Comment:
Previously this used a couch view to find a supply point case.  I preferred this way because in general I assume `db.get()` is faster than `db.view()`, but also because I wanted the location model to control the association between itself and its supply-point case.